src/data_request.rs

```diff
@@ -53,9 +55,43 @@ pub struct DataRequest {
     pub images: bool,
 }
 
+/// A type describing which layers to load.
+#[derive(Clone, Debug, Default, PartialEq, Eq)]
+pub(crate) struct LayerFilter {
+    all: bool,
+    load_default: bool,
+    named_layers: Vec<String>,
+}
+
+impl LayerFilter {
+    fn from_bool(b: bool) -> Self {
+        LayerFilter { all: b, ..Default::default() }
+    }
+
+    pub(crate) fn should_load(&self, name: &str, path: &Path) -> bool {
+        dbg!(name, path);
+        self.all
+            || (self.load_default && path == Path::new("glyphs"))
+            || self.named_layers.iter().any(|req_name| name == req_name)
```

Comment:
I believe it was deliberate because

```rust
let v = [String::from("hello"), String::from("world")]; // slice of `String`
assert!(v.contains("hello"));  // would fail because "expected struct `String`, found `str`"
```

The Vec::contains docs mention this too
https://doc.rust-lang.org/std/vec/struct.Vec.html#method.contains

Reply:
there's a open issue on that in https://github.com/rust-lang/rust/issues/42671, pity that Vec::contains doesn't just Do The Right Thing™

Reply:
yea, I tried this with contains but ran into the issue described. :)